Lib/fontTools/varLib/__init__.py

```diff
@@ -203,6 +203,54 @@ def _SetCoordinates(font, glyphName, coord):
 	font["hmtx"].metrics[glyphName] = horizontalAdvanceWidth, leftSideBearing
 
 
+def _compressedDelta(delta, control):
+	# Try to optimize the deltas for each contour.
+	# E.g. if a whole contour is moved, only one delta needs to be stored.
+	# The result is not always smaller than storing the deltas verbatim,
+	# because it requires that the point index for each of the remaining
+	# deltas is stored.
+	contour_deltas = []
+	i = 0
+	if type(control[0]) == str:
+		# Composite glyph: control contains the component names
+		# Apparently, probably depending on the component flags, the delta
+		# record already contains the sidebearing points in some cases, but
+		# not in others.
+		contour_deltas.extend([[d] for d in delta[:len(control)]])
+		if len(delta) > len(control):
+			# Phantom points are included in the delta record, add them to the contour deltas.
+			# Phantom points are not interpolated, they can each be optimized separately.
+			contour_deltas.extend([[d] for d in delta[-4:]])
+		assert len(contour_deltas) == len(delta)
+	else:
+		# Simple glyph: control contains the list of contour end points and flags
+		for j in control[0]:
+			# Store each contour in a sublist of contour_deltas
+			contour_deltas.append(delta[i:j + 1])
+			i = j + 1
+		# Phantom points are not interpolated, they can each be optimized separately.
+		contour_deltas.extend([[d] for d in delta[-4:]])
+	compressed_delta = []
+	for contour_delta in contour_deltas:
+		delta_set = set(contour_delta)
+		if len(delta_set) == 1:
+			# The whole contour is moved, just store one delta value for the start point.
+			first_delta = list(delta_set)[0]
```

Comment:
hm, this line looks suspicious `list(delta_set)[0]`, as sets are unsorted (it's order is implementation specific and on python 3 uses a randomized seed), thus the `first_delta` will be different at every run.